src/engine/sync/basesyncablelistener.cpp

```diff
@@ -184,3 +184,40 @@ void BaseSyncableListener::checkUniquePlayingSyncable() {
         unique_syncable->notifyOnlyPlayingSyncable();
     }
 }
+
+void BaseSyncableListener::setPlayingSyncable(Syncable* pSource) {
+    Syncable *currentSyncable = pSource;
+    // Sort the syncables with the playing syncables clustered in the beginning
+    // and appending the current syncable at the end of the playing syncables.
+    // This allows for keeping track of the first played deck, which is needed 
```

Comment:
Here we should choose the current playing deck. 
It can be queried via PlayerInfo::getCurrentPlayingDeck()

Reply:
This is an interesting idea. Using this, We won't need to keep track of the time of playing of decks. And it makes more sense to set the most audible deck as master if sync is turned off.

Reply:
To implement this we'll have to create more listeners in EngineSync i guess. Otherwise it won't be able to update the master bpm on volume change.

Reply:
The PlayerInfo class does this already.
Not sure if it meets you timing requirements though.
I would prefer to keep the current track logic in a single place.

Reply:
Yes, It does not meet the timing requirements currently. Will need to implement more listeners for volume change & crossfader updates.

> I would prefer to keep the current track logic in a single place.

Can you elaborate?

Reply:
Currently the PlayerInfo Class is resposible to decide which is the "current deck" and "current track". 
It would be nice to re-use it here. If it is not possible for some reasons it is also OK.

Reply:
I was trying to add slots for volume level updates. But for some reason, they are not getting connected. `valueChanged()` signal is not emitted although the key "volume" is getting updated for that group. I checked this in the developer tools.

Can you review? I have pushed a commit for the same.